example-projects/rdkafka-example/src/main.rs

```diff
@@ -0,0 +1,249 @@
+        //.set("auto.offset.reset", "smallest")
+        .set_log_level(RDKafkaLogLevel::Debug)
+        .create_with_context(context)
+        .expect("Consumer creation failed");
+
+    consumer
+        .subscribe(&topics.to_vec())
+        .expect("Can't subscribe to specified topics");
+
+    // consumer.start() returns a stream. The stream can be used ot chain together expensive steps,
+    // such as complex computations on a thread pool or asynchronous IO.
+    let mut message_stream = consumer.start();
+
+    while let Some(message) = message_stream.next().await {
+        match message {
+            Err(e) => warn!("Kafka error: {}", e),
+            Ok(m) => {
+                let event = m.into_event().unwrap();
+                info!("Received Event: {:#?}", event);
+
+                consumer.commit_message(&m, CommitMode::Async).unwrap();
+            }
+        };
+    }
+}
+
+#[tokio::main]
+async fn consumer_example(log: Option<&str>, brokers: &str, group_id: &str, topics: Vec<&str>) {
+    setup_logger(true, log);
+
+    let (version_n, version_s) = get_rdkafka_version();
+    info!("rd_kafka_version: 0x{:08x}, {}", version_n, version_s);
+
+    consume_and_print(brokers, group_id, &topics).await
+}
+
+async fn produce(brokers: &str, topic_name: &str) {
+    let producer: &FutureProducer = &ClientConfig::new()
+        .set("bootstrap.servers", brokers)
+        .set("message.timeout.ms", "5000")
+        .create()
+        .expect("Producer creation error");
+
+    // This loop is non blocking: all messages will be sent one after the other, without waiting
+    // for the results.
+    let futures = (0..5)
+        .map(|i| async move {
+            // The send operation on the topic returns a future, which will be
+            // completed once the result or failure from Kafka is received.
+            let event = EventBuilderV10::new()
+                .id("0001")
+                .ty("example.test")
+                .source(Url::from_str("http://localhost/").unwrap())
+                .data("application/json", json!({"hello": "world"}))
+                .extension("someint", "10")
+                .build()
+                .unwrap();
+
+            info!("Sending event: {:#?}", event);
+
+            let delivery_status = producer
+                .send(
+                    FutureRecord::to(topic_name)
+                        .event(&event.serialize_event().unwrap())
+                        .unwrap()
+                        .key(&format!("Key {}", i)),
+                    0,
+                )
+                .await;
+
+            // This will be executed when the result is received.
+            info!("Delivery status for message {} received", i);
+            delivery_status
+        })
+        .collect::<Vec<_>>();
+
+    // This loop will wait until all delivery statuses have been received.
+    for future in futures {
+        info!("Future completed. Result: {:?}", future.await);
+    }
+}
+
+#[tokio::main]
```

Comment:
should tokio main macro annotate main function?

Reply:
silly copy paste mistake :P